components/common-go/log/log.go

```diff
@@ -131,28 +132,40 @@ func (f *gcpFormatter) Format(entry *log.Entry) ([]byte, error) {
 	return f.JSONFormatter.Format(entry)
 }
 
-// Writer produces a writer that wraps everything that's written to it in a log message.
+// JSONWriter produces a writer that wraps everything
+// that's written to it in a log message.
 // Callers are expected to close the returned writer.
 //
-// Beware: due to logEntry not being synchronised, writes to the returned writer must not
-//         concurrent.
-func Writer(logEntry *logrus.Entry) io.WriteCloser {
+// Beware: due to logEntry not being synchronised, writes
+// to the returned writer must not concurrent.
+func JSONWriter(logEntry *logrus.Entry) io.WriteCloser {
 	rd, rw := io.Pipe()
 	go func() {
-		b := make([]byte, 4096)
+		dec := json.NewDecoder(rd)
 		for {
-			n, err := rd.Read(b)
-			if err == io.EOF {
-				break
+			var entry jsonEntry
+			if err := dec.Decode(&entry); err != nil {
+				continue
```

Comment:
In this case we'd be swallowing log messages. We might want to fall back to the "messy" log format here.

Reply:
done